File: maths/cycle1/fiches-methodes3e/I/Pascal/m0207.pas
program m0207;

{$mode objfpc}{$H+}

uses
    method02, display;

var
    eggs, capacity, boxes, remainder : Byte;
    bool : Boolean;

begin
    WriteLn('# Exercice 7.');
    WriteLn('# ----------');
    WriteLn('# Une fermière a récolté 59 œufs. Pour les transporter au marché et les vendre,');
    WriteLn('# elle les range dans des boîtes de 12. Combien lui faudra-t-il de boîtes ?');
    WriteLn('-----------------------------------------------------------------------------------');

    bool := DisplayMethodWording;
    if bool then PutAMethod();

    bool := PutUpCorrectVersion;
    if not bool then exit;

    eggs := 59;
    capacity := 12;
    boxes := eggs div capacity;
    remainder := eggs mod capacity;

    WriteLn(eggs, ' = ', capacity, ' x ', boxes, ' + ', remainder, '.');
    if (remainder > 0) then
    begin
        WriteLn('Il faudra ', boxes + 1, ' boîtes à la fermière pour ranger ses œufs.');
        Write('La dernière boîte contiendra ', remainder);
        if (remainder > 1) then
            WriteLn(' œufs.')
        else
            WriteLn(' œuf.');
    end
    else
        WriteLn('Les œufs seront contenus exactement dans ', boxes, ' boîtes.');

end.

File: maths/cycle1/fiches-methodes3e/I/Ruby/m0207.rb
#!/usr/bin/env ruby

require_relative "method02"
require_relative "corrective"

class Solution < Method02
  include Corrective

  def answering
    eggs = 59
    capacity = 12
    boxes = eggs / capacity
    remainder = eggs % capacity

    puts "#{eggs} = #{capacity} x #{boxes} + #{remainder}."
    if remainder > 0 then
      puts "Il faudra #{boxes + 1} boîtes à la fermière pour ranger ses œufs."
      print "La dernière boîte contiendra #{remainder} "
      puts remainder > 1 ? 'œufs.' : 'œuf.'
    else
      puts "Les œufs seront contenus exactement dans #{boxes} boîtes."
    end
  end
end

s = Solution.new
s.exercise_07
bool = true
bool = s.put_up_method?
if bool then s.put_up_method end
s.put_up_correct_version?
s.answering

